tests/Utils/SchemaPrinterTest.php

```diff
@@ -148,6 +148,67 @@ public function testPrintNonNullArrayNonNullStringField() : void
         );
     }
 
+    /**
+     * @see it('Prints Field With "@deprecated" Directive')
+     *
+     * @dataProvider deprecationReasonDataProvider
+     */
+    public function testPrintDeprecatedField($deprecationReason, string $expectedDeprecationDirective) : void
+    {
+        $output = $this->printSingleFieldSchema([
+            'type' => Type::int(),
+            'deprecationReason' => $deprecationReason,
+        ]);
+        self::assertSame(
+            '
+type Query {
+  singleField: Int'.$expectedDeprecationDirective.'
+}
+',
+            $output
+        );
+    }
+
+    public function deprecationReasonDataProvider(): \Generator
```

Comment:
`deprecationReason` is typed as `string|null` so there's no need to test for other types like int or array.

Reply:
> /** @var string|null */

is "typed" 😄

Reply:
Yes. We have static analysis and CI to guard that.